addon/components/location-plugin/map.gts

```diff
@@ -0,0 +1,131 @@
+import { type LeafletMouseEvent } from 'leaflet';
+import { restartableTask } from 'ember-concurrency';
+import { task as trackedTask } from 'ember-resources/util/ember-concurrency';
+import AuLoader from '@appuniversum/ember-appuniversum/components/au-loader';
+import { Address } from '@lblod/ember-rdfa-editor-lblod-plugins/plugins/location-plugin/utils/address-helpers';
+import {
+  convertLambertCoordsToWGS84,
+  GeoPos,
+  type GlobalCoordinates,
+} from '@lblod/ember-rdfa-editor-lblod-plugins/plugins/location-plugin/utils/geo-helpers';
+
+export type LocationType = 'address' | 'place';
+
+const COORD_SYSTEM_CENTER: GlobalCoordinates = {
+  lng: 4.450822554431,
+  lat: 50.50526730529,
+};
+
+function displayLocation(location: GeoPos | undefined) {
+  return location ? `[x: ${location.lambert.x}, y: ${location.lambert.y}]` : '';
+}
+
+interface Signature {
+  Args: {
+    address: Address | undefined;
+    locationType: LocationType;
+    location?: GeoPos;
+    setLocation: (loc: GlobalCoordinates) => void;
+    existingLocation?: GeoPos;
+  };
+  Element: HTMLElement;
+}
+
+async function ensureGlobalCoordinates(geoPos: GeoPos) {
+  if (geoPos.global) {
+    return geoPos.global;
+  }
+  return await convertLambertCoordsToWGS84(geoPos.lambert);
+}
+
+export default class LocationPluginMapComponent extends Component<Signature> {
+  @tracked mapCenter = COORD_SYSTEM_CENTER;
+  @tracked zoom = 7;
+  @tracked existingLocationCoords: GlobalCoordinates | undefined;
+
+  addressLatLngLookup = restartableTask(
+    async (): Promise<GlobalCoordinates | undefined> => {
+      const { address, existingLocation } = this.args;
+      if (address) {
+        this.mapCenter = await ensureGlobalCoordinates(
+          address.location.location,
+        );
+        this.zoom = 19;
+      } else if (existingLocation) {
+        this.mapCenter = await ensureGlobalCoordinates(existingLocation);
+        this.existingLocationCoords = this.mapCenter;
+        this.zoom = 19;
+      }
+      return COORD_SYSTEM_CENTER;
+    },
+  );
+  addressLatLng = trackedTask<GlobalCoordinates | undefined>(
+    this,
+    this.addressLatLngLookup,
+    () => [this.args.address, this.args.existingLocation],
+  );
+
+  get doRenderMap() {
+    return this.addressLatLng.value && this.zoom;
+  }
+
+  @action
+  onMapClick(event: LeafletMouseEvent) {
+    if (this.args.locationType === 'place') {
+      this.args.setLocation(event.latlng);
+    }
+  }
+
+  <template>
+    <div class='map-wrapper' ...attributes>
+      {{#if this.addressLatLng.isRunning}}
+        <AuLoader @hideMessage={{true}}>
+          {{t 'common.initialBounds.loading'}}
+        </AuLoader>
+      {{else if this.addressLatLng.value}}
+        <LeafletMap
+          @center={{this.mapCenter}}
+          @zoom={{this.zoom}}
+          @onClick={{this.onMapClick}}
+          as |layers|
+        >
+          <layers.tile
+            @url='https://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png'
```

Comment:
_Non-blocking_: why the "[Humanitarian map style](https://wiki.openstreetmap.org/wiki/Humanitarian_map_style)" instead of the regular one `https://tile.openstreetmap.org/{zoom}/{x}/{y}.png`?

Reply:
P.S. Also using OSM tiles requires license attribution, right now it only shows the "Leaflet" link. Not sure how easily is that done with `ember-leaflet` 

![CleanShot 2024-06-12 at 17 27 37@2x](https://github.com/lblod/ember-rdfa-editor-lblod-plugins/assets/769698/3fb67690-fdb9-4b8d-818e-130d18acdb53)

Reply:
Seemed cleaner... Maybe regular OSM is better for contrast though.